modyn/selector/internal/selector_strategies/downsampling_strategies/scheduler_downsampling_strategy.py

```diff
@@ -0,0 +1,97 @@
+from modyn.selector.internal.selector_strategies.downsampling_strategies import AbstractDownsamplingStrategy
+from modyn.selector.internal.selector_strategies.downsampling_strategies.utils import instantiate_downsampler
+
+
+class SchedulerDownsamplingStrategy(AbstractDownsamplingStrategy):
+    def __init__(self, downsampling_config: dict, maximum_keys_in_memory: int):
+        if "first_strategy" not in downsampling_config:
+            raise ValueError(
+                "If you want to use a Downsampling Scheduler, please specify the first and the second strategy."
+            )
+        first_strategy = downsampling_config["first_strategy"]
+
+        if "second_strategy" not in downsampling_config:
+            raise ValueError(
+                "If you want to use a Downsampling Scheduler, please specify the first and the second strategy."
+            )
+        second_strategy = downsampling_config["second_strategy"]
+
+        if "threshold" not in downsampling_config:
+            raise ValueError("Please specify after how many epochs you want to switch strategy.")
+        self.threshold = downsampling_config["threshold"]
+
+        if "ratio" in downsampling_config:
+            raise ValueError(
+                "SchedulerDownsamplingStrategy has no downsampling ratio. Specify the downsampling ratio"
+                "inside first_strategy and second_strategy."
+            )
+        downsampling_config["ratio"] = 100  # just to deal with super
+
+        if "sample_then_batch" in downsampling_config:
+            raise ValueError("SchedulerDownsamplingStrategy has no downsampling mode (sample_then_batch parameter).")
```

Comment:
maybe be more explicit (you can make it like "if ratio in downsampling_config or if stb in config or ...: ValueError(please specify the strategy configs within the strategy objects, not on scheduler level)" or sth like that